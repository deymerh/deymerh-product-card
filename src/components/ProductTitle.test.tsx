import React from 'react';
import renderer from 'react-test-renderer';

import { ProductTitle, ProductCard} from '../components';
import { product1 } from '../data/products';

describe('Test in <ProductTitle />', () => {
    test('should to show the component correctly', () => {
      const wrapper = renderer.create(
        <ProductTitle title='Custom test title' />
      )
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should to show the product name from app context', () => {
      const wrapper = renderer.create(
        <ProductCard
          product={product1}
        >
          {
            ()=>(
              <ProductTitle />
            )
          }
        </ProductCard>
      )
      expect(wrapper.toJSON()).toMatchSnapshot();
    })
})