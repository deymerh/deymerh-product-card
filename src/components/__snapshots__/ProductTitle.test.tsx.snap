// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductTitle /> should to show the component correctly 1`] = `
<span
  className="productDescription
            undefined"
>
  Custom test title
</span>
`;

exports[`Test in <ProductTitle /> should to show the product name from app context 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription
            undefined"
  >
    Coffe Mug - Card sin imagen
  </span>
</div>
`;
